go/activity.go

```diff
@@ -0,0 +1,37 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+func BookHotel(ctx context.Context, bookingInfo BookingInfo, start time.Time, end time.Time, idempotencyKey string) error {
```

Comment:
We generally recommend users [pass a single struct vs lots of input parameters ](https://legacy-documentation-sdks.temporal.io/go/develop-activities#parameters)